### ISSUE TITLE
pre-flight checks for pipelines

### ISSUE THREAD

**Issue by joaquinvanschoren** [openml/openml-python#211]
Hi, 

It can sometimes be annoying when you train a pipeline and then want to publish it, but the server tells you that you can't (e.g. because you use the same component twice in a pipeline).

Would it be hard to add a checking routine to check the pipeline, so that people can check compatibility before training and uploading the pipeline?

Cheers,
Joaquin

**Comment by janvanrijn** [comment]
One additional comment kind of related to this. When testing, apparently the flow Bagging(Bagging(DecisionTree)) is allowed. This makes sense, as it contains of three distinct components:
* Bagging(Bagging(DecisionTree))
* Bagging(DecisionTree)
* DecisionTree

Should also check whether sklearn handles this flow correctly (wrt parameter settings)

**Comment by janvanrijn** [comment]
One more comment: This is what is currently happening. When uploading the following flow:
`illegal = sklearn.pipeline.Pipeline(steps=[('imputer1', sklearn.preprocessing.Imputer()),
                                                   ('imputer2', sklearn.preprocessing.Imputer()),
                                                   ('classif', sklearn.tree.DecisionTreeClassifier())])
`

the error that I get is the following:

`ValueError: Found a second occurence of component sklearn.preprocessing.imputation.Imputer when trying to serialize Pipeline(steps=[('imputer1', Imputer(axis=0, copy=True, missing_values='NaN', strategy='mean', verbose=0)), ('imputer2', Imputer(axis=0, copy=True, missing_values='NaN', strategy='mean', verbose=0)), ('classif', DecisionTreeClassifier(class_weight=None, criterion='gini', max_depth=None,
            max_featu...it=2, min_weight_fraction_leaf=0.0,
            presort=False, random_state=None, splitter='best'))]).
`
Raised by the openml-python package. Furthermore, the openml-python package also relies on the property of flows being unique (sklearn_converter.py:235).

I also changed the order of run_task execution in a previous fix, which should ensure that before the flow is ran, this error is given (I don't recall exactly why we did this the other way around, maybe @mfeurer can enlighten this)

Added testcase, case closed?
